utils/s2n_random.c

```diff
@@ -512,11 +512,22 @@ int s2n_openssl_compat_status(void)
     return 1;
 }
 
+#if S2N_LIBCRYPTO_SUPPORTS_ENGINE
```

Comment:
What methods are you gating here? Are they present in the feature probe?

Reply:
- One of the parameters are `ENGINE *unused`
- RAND_METHOD is gated due to difference between openssl and awslc